src/parsing/internal-parsing-logic.ts

```diff
@@ -379,8 +379,18 @@ function convertASTToInstructionSet(
             if (expression[1].includes("|")) {
               acc.instructionSet.push(truKey);
             }
-            // Currently only supporting Memory type need to expand to support placeholder usage in tracker updates
-            acc.instructionSet.push(0);
+            const trackerName = expression[1].split(":")[1];
+            const trackerType = indexMap.find(
+              (mapping) => mapping.name == trackerName
+            )?.type;
+
+            // if trackerType is string(1) or bytes(5) then push 1 indicating placeholder tracker
+            // else push 0 indicating memory tracker
+            if (trackerType === 1 || trackerType === 5) {
```

Comment:
Do we have the enum setup here that we can notify what's what? That would be preferential.

Reply:
updated